gomock/controller.go

```diff
@@ -260,6 +260,13 @@ func (ctrl *Controller) Call(receiver interface{}, method string, args ...interf
 // were called. It should be invoked for each Controller. It is not idempotent
 // and therefore can only be invoked once.
 func (ctrl *Controller) Finish() {
+	// If we're currently panicking, probably because this is a deferred call.
+	// This must be recovered in the deferred function.
+	err := recover()
+	ctrl.finish(false, err)
+}
+
+func (ctrl *Controller) finish(cleanup bool, panicErr interface{}) {
```

Comment:
can panicErr just be type `error`?

Reply:
It can't, it could be anything. It represents whatever what the input to `panic`.: https://github.com/golang/go/wiki/PanicAndRecover#panic

Reply:
Ah ok, makes sense